builders/base/buildBaseBuilder.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+declare -r SCRIPT_DIR=$( cd $( dirname "$0" ) && pwd )
+declare -r REPO_DIR=$( cd $( dirname "$0" ) && cd .. && cd .. && pwd )
+source $REPO_DIR/build/__variables.sh
+
+# constants
+declare -r ORYX_BUILDPACK_IMAGE_PLACEHOLDER="%ORYX_BUILDPACK_IMAGE%"
+declare -r ORYX_BUILDPACK_VERSION_PLACEHOLDER="%ORYX_BUILDPACK_VERSION%"
+declare -r ORYX_RUN_STACK_IMAGE_PLACEHOLDER="%ORYX_RUN_STACK_IMAGE%"
+declare -r ORYX_BUILD_STACK_IMAGE_PLACEHOLDER="%ORYX_BUILD_STACK_IMAGE%"
+declare -r MCR_BUILDER_IMAGE_REPO="mcr.microsoft.com/oryx/builder"
+declare -r MCR_CLI_IMAGE_REPO="mcr.microsoft.com/oryx/cli"
+
+# parameter defaults
+builderImageVersion="20230208.1"
+destinationFqdn="oryxprodmcr.azurecr.io"
+destinationRepo="public/oryx/builder"
+buildpackVersion="0.0.4"
+
+PARAMS=""
+while (( "$#" )); do
+  case "$1" in
+    -c|--cli-builder-image)
+    cliBuilderImage=$2
+    shift 2
+    ;;
+    -v|--builder-image-version)
+    builderImageVersion=$2
+    shift 2
+    ;;
+    -f|--destination-registry-fqdn)
+    destinationFqdn=$2
+    shift 2
+    ;;
+    -r|--destination-registry-repo)
+    destinationRepo=$2
+    shift 2
+    ;;
+    --buildpack-version)
+    buildpackVersion=$2
+    shift 2
+    ;;
+    --) # end argument parsing
+    shift
+    break
+    ;;
+    -*|--*=) # unsupported flags
+    echo "Error: Unsupported flag $1" >&2
+    exit 1
+    ;;
+    *) # preserve positional arguments
+    PARAMS="$PARAMS $1"
+    shift
+    ;;
+  esac
+done
+
+function replaceRepo() {
+  local imageName="$1"
+  local newRepo="$2"
+  # Retag build image with new repo
+    IFS=':' read -ra SPLIT_IMAGE_NAME <<< "$imageName"
+    local repo="${SPLIT_IMAGE_NAME[0]}"
+    local tag="${SPLIT_IMAGE_NAME[1]}"
+    local newImage="$newRepo:$tag"
+    docker tag "$imageName" "$newImage"
+    echo $newImage
+}
+
+if [ -z $cliBuilderImage ]; then
+    cliBuilderImage="$destinationFqdn/public/oryx/cli:builder-debian-buster-$builderImageVersion"
```

Comment:
Could you please clarify why we default to the buster version? Any reasons not to use Bullseye?

Reply:
I defaulted to buster since thats what the current builder is based on. I think that when @cormacpayne originally made it, we didn't have a bullseye CLI image? 

Shouldn't be incredibly difficult to swap out the base OS, probably just a few find and replaces (as long as the tests still pass w/ bullseye). Do you think that would be worthwhile to do?

Reply:
That's correct -- there was no bullseye flavor of the CLI image published out of our repository when the builder was first iterated on. If possible, I think we should migrate to bullseye since it's the only stable version of Debian released (after buster moved to LTS a few months ago).

Reply:
Ok, I'll work on migrating to bullseye. Thanks for the input!